content/post/writing-a-tsdb.md

```diff
@@ -6,52 +6,40 @@ draft = true
 
-we were actually running on graphite+kairosdb for a while. But that only took us so far.
+Plus, I'm sure I've seen about another dozen, ranging from 1-person pet projects to systems supporting the production monitoring of various large companies.
 
+It seems the last thing the world needs is another TSDB.
 
-Some key properties I look for in a TSDB are resource efficiency, performance, a powerful / flexible API and operational simplicity.
-I also want the system to be free / open source, and scale from running some local metrics on my laptop to serving thousands of customer in the datacenter.
+Graphite, despite some well understood limitation and its age, has a tremendously powerful yet simple data processing API. In many ways, its unrivaled. It's the complete package, and it just works. At raintank, we're big fans of Graphite, and it offers a great experience when coupled with Grafana. 
 
-Spoiler alert: I haven't seen a system that checks all the boxes.  Most can get about 4 out of 6.
-There's also no way that I (or we, at raintank) have the resources to build a better general purpose TSDB.  Nor do we want to. We want to build a great monitoring system and reuse existing technology where possible.  We don't want to reinvent what already exists and introduce yet another system that scatters the ecosystem and confuses things.
+Elasticsearch also deseves a spot on the list now. It's no longer just for log searching and events: it's getting [quite good for timeseries metrics](https://www.elastic.co/blog/elasticsearch-as-a-time-series-data-store). CERN, the guys who run the Large Hydron Collider, use it to [monitor their jobs!](http://cds.cern.ch/record/2011172/files/LHCb-TALK-2015-060.pdf)
+
+SPOILER ALERT: Nothing we've mentioned here meets our needs for a hosted TSDB. We want something that (1) is performant and resource efficient (2) completely open source (3) have a powerful API, ideally similar to Graphite (4) extremely stable (5) didn't have a lot of dependencies we weren't comfortable with. No system we've found can check all the boxes. 
+
+We were running KairosDB for a while. We even wrote a [Graphite Finder](http://www.github.com/raintank/) for it. There is now full support for KairosDB in Grafana. At our scale though, it only got us so far. We liked what we saw in Cassandra.
+
+There's no way that we have the resources or the interest to build a better general purpose TSDB from the ground up. Nor do we want to. We want to build a great OpenSaaS platform, and stand on the shoulders of open source giants where possible. If it already exists, we should use it.
 
 # So we didn't want to build yet another TSDB, than why did we?
 
-While with Grafana, we have the interests of a broad group of people at heart, for the hosted metrics platform specifically our requirements are actually fairly narrow:
+As with most things, it starts as a prototype.
```

Comment:
any particular reason you removed the grafana <-> hosted metrics use case separation?
i felt like it was appropriate to make this clear.
 @nopzor1200